server/src/main/java/org/elasticsearch/common/settings/ChangeKeyStorePassphraseCommand.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.settings;
+
+import joptsimple.OptionSet;
+import org.elasticsearch.cli.EnvironmentAwareCommand;
+import org.elasticsearch.cli.ExitCodes;
+import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.cli.UserException;
+import org.elasticsearch.env.Environment;
+
+import java.util.Arrays;
+
+public class ChangeKeyStorePassphraseCommand extends EnvironmentAwareCommand {
+
+
+    ChangeKeyStorePassphraseCommand() {
+        super("Changes the passphrase of a keystore");
+    }
+
+    @Override
+    protected void execute(Terminal terminal, OptionSet options, Environment env) throws Exception {
+        char[] passphrase = null;
+        try {
+            KeyStoreWrapper keystore = KeyStoreWrapper.load(env.configFile());
```

Comment:
this block seems to be duplicated quite a bit and I think we'd benefit from sharing this block of code

Reply:
I gave this a go but I can't seem to come up with a good idea.

- Tried with a `readOrCreate` method but that would either need 
  - to be passed the passphrase in order to decrypt the existing keystore if there. The passphrase is needed in the `execute()` method of the commands also, as we need to encrypt the keystore (with `save()`) after we set the configuration value.
  - check the returned keystore to see if it is a new or an existing keystore and then conditionally read the passphrase. <- Then I'd have to duplicate this part of the code in every place the existing block lies, so not much of an enhancement. 

I'm open to suggestions

Reply:
thanks for trying; seems like it might even make things messier

Reply:
I think this can be done by creating a pojo used by the CLIs which contains the keystore and password? Then the readOrCreate can always return the passphrase (even if empty because we read a keystore without a passphrase).

Reply:
Thanks Ryan , I'll give this a try